crates/sui-source-validation-service/src/lib.rs

```diff
@@ -370,22 +373,35 @@ pub async fn verify_packages(config: &Config, dir: &Path) -> anyhow::Result<Netw
 pub async fn watch_for_upgrades(
     packages: Vec<PackageSource>,
     app_state: Arc<RwLock<AppState>>,
+    network: Network,
     channel: Option<Sender<SuiTransactionBlockEffects>>,
 ) -> anyhow::Result<()> {
     let mut watch_ids = ArrayParams::new();
+    let mut num_packages = 0;
     for s in packages {
         let packages = match s {
             PackageSource::Repository(RepositorySource { packages, .. }) => packages,
             PackageSource::Directory(DirectorySource { packages, .. }) => packages,
         };
         for p in packages {
             if let Some(id) = p.watch {
+                num_packages += 1;
                 watch_ids.insert(TransactionFilter::ChangedObject(id))?
             }
         }
     }
 
-    let client: WsClient = WsClientBuilder::default().build(LOCALNET_WS_URL).await?;
+    let websocket_url = match network {
+        Network::Mainnet => MAINNET_WS_URL,
+        Network::Testnet => TESTNET_WS_URL,
+        Network::Devnet => DEVNET_WS_URL,
+        Network::Localnet => LOCALNET_WS_URL,
+    };
+
+    let client: WsClient = WsClientBuilder::default()
+        .ping_interval(Duration::from_millis(WS_PING_INTERVAL_MS))
```

Comment:
@amnn btw this is the [`ping_interval`](https://docs.rs/jsonrpsee-core/0.20.2/jsonrpsee_core/client/async_client/struct.ClientBuilder.html#method.ping_interval) rust method I use to keep-alive the websocket, and appears to work well empirically (tested on a different server). Without this the connection tends to drop after 30 seconds.